context/app/static/js/shared-styles/charts/VerticalStackedBarChart/VerticalStackedBarChart.jsx

```diff
@@ -10,6 +10,20 @@ import { trimStringWithMiddleEllipsis } from 'js/shared-styles/charts/utils';
 import StackedBar from 'js/shared-styles/charts/StackedBar';
 import VerticalChartGridRowsGroup from 'js/shared-styles/charts//VerticalChartGridRowsGroup';
 
+function TickComponentWithHandlers({ handleMouseEnter, handleMouseLeave }) {
+  // use a callback to avoid creating a new component on every render
+  return useCallback(
+    ({ formattedValue, ...tickProps }) => {
+      return (
+        <Text onMouseEnter={handleMouseEnter({ key: formattedValue })} onMouseLeave={handleMouseLeave} {...tickProps}>
+          {trimStringWithMiddleEllipsis(formattedValue)}
+        </Text>
+      );
+    },
+    [handleMouseEnter, handleMouseLeave],
+  );
+}
+
```

Comment:
This pattern seems strange, was this the outcome of an eslint suggestion? Which rule?